fix(doctor): missing_slice_dir false positive persists via legacy-parser path — parseLegacyRoadmap never sets .pending

## Problem

After PR #2451 (merged 2026-03-25) added `if (slice.pending) continue` guards to skip `missing_slice_dir` / `missing_tasks_dir` checks for unstarted future slices, the false positives persist whenever the **legacy parser fallback** is used instead of the DB path.

Doctor still emits 4 `severity: \"error\"` entries for future slices (S02–S05 here) even though those slices are simply unplanned — their directories are created lazily by `ensurePreconditions()` at dispatch time.

## Root Cause

`doctor.js:440–453` — the DB and legacy branches set different slice shapes:

```js
if (isDbAvailable()) {
    const dbSlices = getMilestoneSlices(milestoneId);
    slices = dbSlices.map(s => ({
        id: s.id,
        title: s.title,
        done: s.status === "complete",
        pending: s.status === "pending",   // ✅ set correctly
        ...
    }));
} else {
    slices = parseLegacyRoadmap(roadmapContent).slices;  // ❌ no .pending field
}
```

Then at `doctor.js:536` / `doctor.js:560`:
```js
if (slice.pending) continue;  // never fires for legacy-parsed slices
```

`parseLegacyRoadmap` → `parseRoadmapSlices` (`roadmap-slices.js:136, 173, 250`) returns objects shaped as:
```ts
{ id, title, risk, depends, done, demo }
```
— **no `pending` property**. `slice.pending` is `undefined` (falsy), so the guard never fires.

This triggers in practice when:
- The project root `gsd.db` is empty or completely unpopulated (e.g. new milestone planned but DB not yet written, or DB deleted/zero-bytes after worktree sync)
- The worktree's `gsd.db` is zero-bytes (deleted by `syncProjectRootToWorktree` and failed to rebuild — see related issue on empty worktree DB)

## Expected Behavior

The legacy-parser fallback path should also skip `missing_slice_dir` / `missing_tasks_dir` for slices that haven't been started. The simplest fix:

```js
// doctor.js:452-453
} else {
    slices = parseLegacyRoadmap(roadmapContent).slices.map(s => ({
        ...s,
        // Legacy parser has no "pending" concept — infer: any non-done slice
        // that has no directory on disk yet is effectively pending/unstarted.
        pending: !s.done,
    }));
}
```

Or alternatively, the guard inside the missing-dir check could be broadened to also skip when `!slice.done` (all undone slices are candidates for being unstarted), matching the intent of the original #2323 fix.

## Environment

- GSD version: 2.46.0
- Model: claude-opus-4-6
- Unit: research-slice / active milestone S01
- Milestone: 5 slices; only S01 started

## Reproduction Context

1. `plan-milestone` completes, creating a roadmap with 5 slices
2. `research-slice M012/S01` is dispatched (first slice)
3. During or after execution, `gsd.db` at project root has 0 rows for M012 (empty/unpopulated DB)
4. Doctor runs → falls back to `parseLegacyRoadmap` → slices lack `.pending` field → S02–S05 each get `missing_slice_dir` severity `"error"`

## Forensic Evidence

```
[ERROR] doctor-issue: Doctor: Missing slice directory for M012/S02
  Code: missing_slice_dir, Scope: slice, Unit: M012/S02, File: .gsd/milestones/M012/slices/S02
[ERROR] doctor-issue: Doctor: Missing slice directory for M012/S03
[ERROR] doctor-issue: Doctor: Missing slice directory for M012/S04
[ERROR] doctor-issue: Doctor: Missing slice directory for M012/S05
```

All 4 errors are for slices that have never been dispatched — directories are expected to not exist yet.

Doctor history confirmed running post-milestone-plan (17:02–17:13 UTC 2026-03-25).

**Related:** Closes #2323 incomplete — PR #2451 fixed the DB path but not the legacy fallback.

---
*Auto-generated by \`/gsd forensics\`*

--- a/src/resources/extensions/gsd/doctor.ts
+++ b/src/resources/extensions/gsd/doctor.ts
@@ -487,7 +487,15 @@ export async function runGSDDoctor(basePath: string, options?: { fix?: boolean; 
         demo: s.demo,
       }));
     } else {
-      slices = parseLegacyRoadmap(roadmapContent).slices;
+      const activeMilestoneId = state.activeMilestone?.id;
+      const activeSliceId = state.activeSlice?.id;
+      slices = parseLegacyRoadmap(roadmapContent).slices.map(s => ({
+        ...s,
+        // Legacy roadmaps only encode done vs not-done. For doctor's
+        // missing-directory checks, treat every undone slice except the
+        // current active slice as effectively pending/unstarted.
+        pending: !s.done && (milestoneId !== activeMilestoneId || s.id !== activeSliceId),
+      }));
     }
     // Wrap in Roadmap-compatible shape for detectCircularDependencies
     const roadmap = { slices };

--- a/src/resources/extensions/gsd/tests/doctor-fixlevel.test.ts
+++ b/src/resources/extensions/gsd/tests/doctor-fixlevel.test.ts
@@ -15,6 +15,7 @@ import { tmpdir } from "node:os";
 import test from "node:test";
 import assert from "node:assert/strict";
 import { runGSDDoctor } from "../doctor.ts";
+import { closeDatabase } from "../gsd-db.ts";
 
 function makeTmp(name: string): string {
   const dir = join(tmpdir(), `doctor-fixlevel-${name}-${Date.now()}-${Math.random().toString(36).slice(2)}`);
@@ -112,6 +113,70 @@ test("fixLevel:all — no reconciliation issue codes are reported", async (t) =>
   assert.ok(roadmapContent.includes("- [ ] **S01"), "roadmap should remain unchecked");
 });
 
+test("legacy roadmap fallback: future slices are treated as pending, active slice is not", async (t) => {
+  const tmp = makeTmp("legacy-pending-fallback");
+  t.after(() => {
+    try { closeDatabase(); } catch { /* noop */ }
+    rmSync(tmp, { recursive: true, force: true });
+  });
+
+  // Force the legacy parser branch.
+  try { closeDatabase(); } catch { /* noop */ }
+
+  const gsd = join(tmp, ".gsd");
+  const m = join(gsd, "milestones", "M001");
+  const s01 = join(m, "slices", "S01", "tasks");
+  mkdirSync(s01, { recursive: true });
+
+  writeFileSync(join(m, "M001-ROADMAP.md"), `# M001: Test
+
+## Slices
+
+- [x] **S01: Done Slice** \`risk:low\` \`depends:[]\`
+  > Done
+- [ ] **S02: Active Slice** \`risk:medium\` \`depends:[S01]\`
+  > In progress
+- [ ] **S03: Future Slice** \`risk:low\` \`depends:[S02]\`
+  > Later
+- [ ] **S04: Future Slice Two** \`risk:low\` \`depends:[S03]\`
+  > Later
+`);
+
+  writeFileSync(join(m, "slices", "S01", "S01-PLAN.md"), `# S01: Done Slice
+
+**Goal:** done
+
+## Tasks
+
+- [x] **T01: Done task** \`est:5m\`
+`);
+
+  // Active slice exists in state/registry but has no directory yet — this should
+  // still be reported as a real error, while future untouched slices should be skipped.
+  const report = await runGSDDoctor(tmp, { scope: "M001" });
+  const missingSliceDirUnits = report.issues
+    .filter(i => i.code === "missing_slice_dir")
+    .map(i => i.unitId)
+    .sort();
+
+  assert.deepStrictEqual(
+    missingSliceDirUnits,
+    ["M001/S02"],
+    "legacy fallback should only report the active slice, not future unstarted slices",
+  );
+
+  const missingTasksDirUnits = report.issues
+    .filter(i => i.code === "missing_tasks_dir")
+    .map(i => i.unitId)
+    .sort();
+
+  assert.deepStrictEqual(
+    missingTasksDirUnits,
+    [],
+    "future slices without directories should be skipped before missing_tasks_dir checks",
+  );
+});
+
 test("fixLevel:all — delimiter_in_title still fixable", async (t) => {
   const tmp = makeTmp("delimiter-fix");
   t.after(() => rmSync(tmp, { recursive: true, force: true }));
@@ -141,7 +206,6 @@ test("fixLevel:all — delimiter_in_title still fixable", async (t) => {
 
   const report = await runGSDDoctor(tmp, { fix: true });
 
-  const delimiterIssues = report.issues.filter(i => i.code === "delimiter_in_title");
   // The milestone-level delimiter is auto-fixed, but the report may or may not include it
   // depending on whether it was fixed successfully. Just verify it ran without crashing.
   assert.ok(report.issues !== undefined, "doctor produces a report");

